### PR TITLE
rn: Document machine-overrides-extender

### PR DESCRIPTION
**Pull request by angolini**
Add a quick explanation on what the `machine-overrides-extender` provides, and a little bit on how it impacts the user.

Please, let me know if there is any other aspect that you think it's important to document, and if anything is confuse.

https://github.com/Freescale/meta-freescale/pull/980

### CODE REVIEW

File: release-notes/source/source-code.rst [hunk 1]
```diff
@@ -13,6 +13,49 @@ The following tree shows the SoC hierarchy:
    :scale: 60%
    :align: center
 
+.. _machine-overrides-extender:
+
+What is machine-overrides-extender
```
Comment (otavio) [inline release-notes/source/source-code.rst:18]:
```suggestion
What is machine-overrides-extender class
```

File: release-notes/source/source-code.rst [hunk 2]
```diff
@@ -13,6 +13,49 @@ The following tree shows the SoC hierarchy:
...
+
+* ``imx-generic-bsp``: Used to configure any generic aspect of the BSP, for example,
+  those configurations that depends only on the MPSoC.
+
+* ``imx-mainline-bsp``: Used to configure the mainline packages.
+
+* ``imx-nxp-bsp``: Used to configure the packages provided by NXP.
+
+The BPS overrides are provided to allow the user to chose between the BSPs and
+using the proper package for each case.
```
Comment (otavio) [inline release-notes/source/source-code.rst:58]: I believe an example using mx8m or similar would be good to add as it shows more details about the process.

**Review by otavio — CHANGES_REQUESTED** [review]

File: release-notes/source/source-code.rst (hunk 2 above)
Comment (angolini) [reply release-notes/source/source-code.rst:58]: @otavio an example on how to add a new machine? Or an example on how to switch from one BSP to the other?

Comment (angolini) [reply release-notes/source/source-code.rst:58]: @otavio , please take a look in the example. My concern is to add a sample code that makes sense now, but will be deprecated in near future, but if you think there is something else to add, please, let me know.

**Review by otavio — APPROVED** [review]

**Pull request merged**